src/init.cpp

```diff
@@ -189,9 +188,9 @@ void Shutdown()
     StopREST();
     StopRPC();
     StopHTTPServer();
-#ifdef ENABLE_WALLET
-    FlushWallets();
-#endif
+    if (g_wallet_init_interface) {
```

Comment:
Can we just make this always non-nullptr and call it directly? The mess of if (g_...) g_...->DoThing() in init.cpp is starting to get very repetitive.

Reply:
FWIW, in my followup, there can be multiple clients, so the if statements turn into for loops. Would be fine to change this, but it's unclear what the benefits would be.